docs/glossary.rst

```diff
@@ -0,0 +1,104 @@
+*********
+Glossary
+*********
+
+For additional terms, refer to the `Jax glossary <https://jax.readthedocs.io/en/latest/glossary.html>`__.
+
+.. glossary::
+
+    Bound Module
+      When a :class:`Module <flax.linen.Module>`
+      is created in a JAX program, it is in an *unbound* state. This means that only
+      dataclass attributes are set, and no variables are bound to the module. When the pure
+      functions :meth:`Module.init() <flax.linen.Module.init>`
+      or :meth:`Module.apply() <flax.linen.Module.apply>`
+      are called, Flax binds the variables to the module, and the module's method code is
+      executed in a bound state, allowing things like calling submodules directly without
+      providing variables. For more details, refer to the
+      `module lifecycle <https://flax.readthedocs.io/en/latest/advanced_topics/module_lifecycle.html>`__.
+
+    Compact / Non-compact Module
+      Modules with a single method are able to declare submodules and variables inline by
+      using the  :func:`@nn.compact <flax.linen.compact>` decorator.
+      These are referred to as “compact-style modules”,
+      whereas modules defining a :meth:`setup() <flax.linen.Module.setup>` method
+      (usually but not always with multiple callable methods)
+      are referred to as “setup-style modules”. To learn more, refer to the
+      `setup vs compact guide <https://flax.readthedocs.io/en/latest/guides/setup_or_nncompact.html>`__.
+
+    `Folding in <https://jax.readthedocs.io/en/latest/_autosummary/jax.random.fold_in.html>`__
+      Generating a new PRNG key given an input PRNG key and integer. Generally faster than
```

Comment:
It is not faster than random.split, but if have an existing RNG `rng` and you want to generate a new one but still be able to use `rng` afterwards, you can do `new_rng = random.fold_in(rng, some_data)` while with `split` you'd have to do `rng, new_rng = random.split(rng)`, which means you effectively create two RNGs, which is slower.